### PR TITLE
feat(spectator): Support input signals

### PR DESCRIPTION
**Pull request by kfrancois**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/spectator/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Types have been cleaned up as `SpectatorDirectiveFactory`, `SpectatorHostFactory` and their related types need some cleanup as the component's generic param is not used everywhere

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #637

## Does this PR introduce a breaking change?
```
[X] Yes
[ ] No
```

`setProps` and `props` were removed from `createHostFactory` and `createDirectiveFactory`. Props should be set through `setHostProps` and `hostProps` instead.

### CODE REVIEW

**Comment by kfrancois** [comment]
This is not ready to merge, see https://github.com/ngneat/spectator/issues/637

File: projects/spectator/test/signal-input/signal-input.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    beforeEach(() => {
+      spectator = createComponent({ props: { show: true } });
+    });
+
+    it('should render a SignalInputComponent', () => {
+      expect(spectator.query('#text')).toContainText('Hello');
+    });
+  });
+
+  describe('with SpectatorHost', () => {
```
Comment (NetanelBasal) [inline projects/spectator/test/signal-input/signal-input.component.spec.ts:21]: Let's add setInput calls here in these tests

**Comment by NetanelBasal** [comment]
LGTM, let's finish it

File: projects/spectator/test/signal-input/signal-input.component.spec.ts (hunk 1 above)
Comment (kfrancois) [reply projects/spectator/test/signal-input/signal-input.component.spec.ts:21]: Good idea, aligned this with the equivalent `jest` test 🚀

**Comment by kfrancois** [comment]
@NetanelBasal I added some cleanup commits, are there any specific changes to docs or other changes you'd like me to make do to ensure this is in a good state to merge? I'd be glad to take them up 😄

**Comment by NetanelBasal** [comment]
We need to update the README to reflect this change:

> setProps and props were removed from createHostFactory and createDirectiveFactory. Props should be set through setHostProps and hostProps instead.

**Comment by kfrancois** [comment]
Great suggestion, thanks! I added some changes to the README but if you want to make any further changes, go ahead 😄 you're much more familiar with the writing style of docs in this repo, so please make any changes you want

**Pull request merged**